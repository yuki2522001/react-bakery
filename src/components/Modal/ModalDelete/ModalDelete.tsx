import React from "react";
import Button from "@components/common/Button/Button/Button";
import "../modal.css";

interface ModalDeleteProps {
  id: string;
  hideModalDelete: () => void;
  deleteProduct: (id: string) => void;
}

const ModalDelete: React.FC<ModalDeleteProps> = ({ id, hideModalDelete, deleteProduct }) => {
  return (
    <div data-testid="modal-delete" className="modal-delete" id="showModal">
      <div className="modal-dialog">
        <div className="modal-content">
          <div className="modal-body">
            <span className="modal-title">Are you sure to delete this product?</span>
          </div>
          <div className="modal-footer">
            <Button text="No" type="warning" onClick={hideModalDelete} />
            <Button text="Yes" type="success" onClick={() => deleteProduct(id)} />
          </div>
        </div>
      </div>
    </div>
  );
};

export default ModalDelete;
