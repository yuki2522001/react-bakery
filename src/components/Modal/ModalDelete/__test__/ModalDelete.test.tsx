import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import ModalDelete from "../ModalDelete";

describe("Modal create component", () => {
  const hideModalDelete = jest.fn();
  const deleteProduct = jest.fn();

  test("should render modal delete component", () => {
    const { getByTestId } = render(
      <ModalDelete
        id={""}
        hideModalDelete={() => {}}
        deleteProduct={() => {}}
      />
    );
    expect(getByTestId("modal-delete")).toBeInTheDocument();
  });

  test("should hide modal delete when click No", () => {
    render(
      <ModalDelete
        id={""}
        hideModalDelete={hideModalDelete}
        deleteProduct={() => {}}
      />
    );
    const hideModal = screen.getByText("No");
    fireEvent.click(hideModal);
    expect(hideModalDelete).toHaveBeenCalled();
  });

  test("should delete product when click Yes", () => {
    render(
      <ModalDelete
        id={""}
        hideModalDelete={() => {}}
        deleteProduct={deleteProduct}
      />
    );
    const hideModal = screen.getByText("Yes");
    fireEvent.click(hideModal);
    expect(deleteProduct).toHaveBeenCalled();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
      <ModalDelete
        id={""}
        hideModalDelete={() => {}}
        deleteProduct={() => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
