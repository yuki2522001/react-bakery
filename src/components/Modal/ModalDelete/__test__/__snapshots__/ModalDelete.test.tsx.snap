// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal create component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal-delete"
    data-testid="modal-delete"
    id="showModal"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-body"
        >
          <span
            class="modal-title"
          >
            Are you sure to delete this product?
          </span>
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn__warning"
          >
            No
          </button>
          <button
            class="btn btn__success"
          >
            Yes
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
